### PR TITLE
[GC-4984] Variables and options must be escaped when echo'd - Batch 2

### PR DESCRIPTION
**Pull request by alexoakland**
### GC-4984
Link to ticket: https://bynder.atlassian.net/browse/GC-4984

### What has been done
- Second lot of fixes for escaping output

**There's a crazy number of changes required to fix all the escaping issues, so will start breaking them up into multiple PRs.**

### CODE REVIEW

File: includes/classes/mapping-post.php [hunk 1]
```diff
@@ -76,31 +77,32 @@ public static function get( $post, $throw_error = false ) {
...
+	 * @since  3.0.0
+	 *
 	 */
 	protected static function get_post( $post ) {
 
 		$post = $post instanceof WP_Post ? $post : get_post( $post );
 
 		if ( ! $post ) {
-			throw new Mapping_Post_Exception( __CLASS__ . ' expects a WP_Post object or post ID.', __LINE__, $post );
+			throw new Mapping_Post_Exception( __CLASS__ . ' expects a WP_Post object or post ID.', __LINE__, wp_kses_post( print_r( $post, true ) ) );
```
Comment (CWDN) [inline includes/classes/mapping-post.php]: Why do we `print_r` as wouldn't that output to the frontend?

Comment (alexoakland) [reply includes/classes/mapping-post.php]: Good spot! My thinking was it was just for reporting as we don't catch the exception anywhere, but, I don't know if I fully trust that it ins't being caught anywhere 😂 

I've whacked the ignore warning comment above to allow it!

File: includes/classes/debug.php [hunk 2]
```diff
@@ -282,7 +280,7 @@ public function do_debug_options_actions( $settings ) {
 			return $orig_settings;
 		}
 
-		wp_die( '<xmp>' . __LINE__ . ') $settings: ' . print_r( $settings, true ) . '</xmp>' . $back_button, __( 'Debug Mode', 'content-workflow-by-bynder' ) );
+		wp_die( '<xmp>' . __LINE__ . ') $settings: ' . esc_html(print_r( $settings, true )) . '</xmp>' . wp_kses_post($back_button), esc_html__( 'Debug Mode', 'content-workflow-by-bynder' ) );
```
Comment (HughbertD) [inline includes/classes/debug.php:283]: @alexoakland Will this escape all the `<pre>` tags?

Comment (alexoakland) [reply includes/classes/debug.php:283]: It will! Good spot! I'll wrap it in them to keep things simple!

**Review by CWDN — APPROVED** [review]

**Pull request merged**